### DISCUSSION TITLE
Variable missing when calling function without a loop in blade

### DISCUSSION THREAD

**Discussion by JoshuaHeathcote1987** [livewire/livewire#2876] [4. General]
Ive been having an issue through the development of this app, which I just ignored until now, which is the time it can no longer be ignored.

The problem is that once the main area is filled, when clicking on ANY other input, which may not even have anything to do with that area, there is an error message displayed.

https://www.youtube.com/watch?v=5K1DGk8in9U

I took a video, and will dump the code.

Blade code wont display so I took a screenshot.

https://ibb.co/K2sTxLF

Livewire code

This code is run when the day, year, teacher is selected.

`    public function getStudents()
    {
        $this->students = Student::with('attendance')->whereHas('attendance', function ($query) {
            $query->where('teacher_id', $this->teacherId);
        })->get();

        $this->attendances = DB::table('attendances')
            ->where('attendances.teacher_id', '=', $this->teacherId)
            ->where('attendances.month', '=', $this->month)
            ->where('attendances.year', '=', $this->year)
            ->get();
    }`

This code is run when a check box is clicked

`    public function addAttendanceRecord($studentId, $day)
    {
        $teacherId = $this->teacherId;
        $studentId = $studentId;
        $day = $day;
        $month = $this->month;
        $year = $this->year;

        try {
            $attendance = Attendance::where('teacher_id', $teacherId)
                ->where('student_id', $studentId)
                ->where('day', $day)
                ->where('month', $month)
                ->where('year', $year)
                ->first();

            $attendance->delete();
        } catch (\Throwable $th) {
            $attendance = Attendance::create([
                'teacher_id' => $this->teacherId,
                'student_id' => $studentId,
                'day' => $day,
                'month' => $this->month,
                'year' => $this->year,
            ]);
        }

    }`

When displaying the table where no checkbox has been click doesn't show this problem. My outlook is that something is clearing the variable on this area, but am not sure.

blade

input class="form-check-input" type="checkbox" wire:click="addAttendanceRecord('{{$student->id}}', '{{$i}}')" 
                                    
                                        @foreach($attendances as $attendance)
                                            @if($student->id == $attendance->student_id && $attendance->day == $i)
                                                checked
                                            @endif
                                        @endforeach

**Comment by gwleuverink** [comment]
The video makes me suspect the data changes when it's rehydrated. Can you try using a computed property instead?

``` php
public function getStudentsProperty()
{
    // return students
    // Don't forget to remove the old $students public property
}
```

Then in your template use the computed property: `{{ $students }}` becomes `{{ $this->students }}`

As a general rule of thumb I try to use public properties only for state that needs to be reactive in some way. Like an input that changes a value, or a button that toggles something. All public state is sent over the wire in each request so I usually avoid putting big lists of things in there unless necessary of course .
